frontend-react/src/components/Admin/MessageTesting.tsx

```diff
@@ -0,0 +1,176 @@
+import { Button, GridContainer, Textarea } from "@trussworks/react-uswds";
+import { ChangeEvent, useState } from "react";
+import { Helmet } from "react-helmet-async";
+import { useParams } from "react-router";
+import { AdminFormWrapper } from "./AdminFormWrapper";
+import { EditReceiverSettingsParams } from "./EditReceiverSettings";
+import useReportTesting from "../../hooks/api/messages/UseMessageTesting";
+import { Icon } from "../../shared";
+import AdminFetchAlert from "../alerts/AdminFetchAlert";
+import Spinner from "../Spinner";
+import Title from "../Title";
+
+function ReportTesting() {
+    const { orgname, receivername } = useParams<EditReceiverSettingsParams>();
+    const { testMessages, isDisabled, isLoading } = useReportTesting();
+    const [selectedOption, setSelectedOption] = useState<string | null>(null);
+    const [currentTestMessages, setCurrentTestMessages] = useState(testMessages);
+    const [openCustomMessage, setOpenCustomMessage] = useState(false);
+    const [customMessageNumber, setCustomMessageNumber] = useState(1);
+
+    if (isDisabled) {
+        return <AdminFetchAlert />;
+    }
+    if (isLoading || !currentTestMessages) return <Spinner />;
+
+    const handleSelect = (event: ChangeEvent<HTMLInputElement>) => {
+        setSelectedOption(event.target.value);
+    };
+
+    const handleAddCustomMessage = () => {
+        setSelectedOption(null);
+        setOpenCustomMessage(true);
+    };
+
+    const CustomMessage = () => {
```

Comment:
This should be broken out into it's own component file taking what is needed as props (may require creating a subdirectory for MessageTesting and moving related files).